### PR TITLE
`Programming exercises`: Change Button Labels and add support for Local Testing and Feedback Mode

### PR DESCRIPTION
**Pull request by Kroko-fant**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
-> no server changes made
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context

https://github.com/ls1intum/Orion/issues/93/

### Description
Orion has received a patch that allows the usage of more IDEs next to IntelliJ. Therefore some of the Button labels weren't up to date anymore.
Furthermore Orion gets an update where students can view inline feedback directly in their IDE. For that a new Reload Button is introduced

### Steps for Testing
In Artemis:
1. Open a programming exercise! no extra button should be there

In [Orion](https://github.com/ls1intum/Orion)
1. Open a programming exercise
3. Test if the buttons are all appearing
2.1. Start assessment button
2.2. Open in your IDE button
2.3. Submit

No changes for Tutor buttons but they still should be visible

Without orion, all orion buttons should not be visible. (can be tested with a normal webbrowser)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

grafik

grafik

### CODE REVIEW

File: src/main/webapp/app/orion/participation/orion-exercise-details-student-actions.component.ts [hunk 1]
```diff
@@ -61,4 +61,16 @@ export class OrionExerciseDetailsStudentActionsComponent implements OnInit {
...
+
+    /**
+     * returns feedback for an exercise
+     */
+    initializeFeedback() {
+        const { results, submissions } = this.exercise.studentParticipations![0] as ProgrammingExerciseStudentParticipation;
+        const feedbacks = results![0].feedbacks;
+        if (submissions) {
+            // @ts-ignore
+            this.orionConnectorService.initializeFeedback(submissions[0].id, feedbacks);
```
Comment (Strohgelaender) [inline src/main/webapp/app/orion/participation/orion-exercise-details-student-actions.component.ts:73]:
```suggestion
        if (submissions?.length) {
            this.orionConnectorService.initializeFeedback(submissions[0].id, feedbacks);
```

File: src/main/webapp/app/orion/assessment/orion-exercise-assessment-dashboard.component.ts [hunk 2]
```diff
@@ -54,4 +55,6 @@ export class OrionExerciseAssessmentDashboardComponent implements OnInit {
     downloadSubmissionInOrion(submission: Submission | 'new', correctionRound = 0, testRun = false) {
         this.orionAssessmentService.downloadSubmissionInOrion(this.exerciseId, submission, correctionRound, testRun);
     }
+
+    protected readonly OrionButtonType = OrionButtonType;
```
Comment (JohannesStoehr) [inline src/main/webapp/app/orion/assessment/orion-exercise-assessment-dashboard.component.ts:59]: We normally add these above all methods

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/orion/management/orion-programming-exercise.component.ts [hunk 3]
```diff
@@ -56,4 +57,6 @@ export class OrionProgrammingExerciseComponent implements OnInit {
             this.orionConnectorService.log(error);
         }
     }
+
+    protected readonly OrionButtonType = OrionButtonType;
```
Comment (Strohgelaender) [inline src/main/webapp/app/orion/management/orion-programming-exercise.component.ts:61]: same here, move this next to the ExerciseView

File: src/main/webapp/app/shared/orion/orion-button/orion-button.component.html [hunk 4]
```diff
@@ -8,6 +8,8 @@
     (click)="handleClick()"
 >
     <fa-icon [hidden]="!buttonLoading" [spin]="true" [icon]="faCircleNotch"></fa-icon>
-    <span *ngIf="!buttonLoading" class="intellij-logo"></span>
+    <fa-icon [hidden]="buttonType !== OrionButtonType.Reload" [icon]="faRotateRight"></fa-icon>
+    <fa-icon [hidden]="buttonType !== OrionButtonType.Submit" [icon]="faUpload"></fa-icon>
+    <fa-icon [hidden]="buttonType !== OrionButtonType.Download" [icon]="faDownload"></fa-icon>
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/orion/orion-button/orion-button.component.html:13]: why do you use `!==` here?

File: src/main/webapp/app/shared/orion/orion-button/orion-button.component.ts [hunk 5]
```diff
@@ -35,4 +46,6 @@ export class OrionButtonComponent {
             this.clickHandler.emit();
         }
     }
+
+    protected readonly OrionButtonType = OrionButtonType;
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/orion/orion-button/orion-button.component.ts:50]: move this next to the icons.

File: src/main/webapp/app/shared/orion/orion-button/orion-button.component.html (hunk 4 above)
Comment (Kroko-fant) [reply src/main/webapp/app/shared/orion/orion-button/orion-button.component.html:13]: the icons only appear when the buttontype is matched as the condition specifies when the icon is hidden

**Review by MaximilianAnzinger — APPROVED** [review]
Tested in testing session

**Review by laurenzfb — DISMISSED** [review]
tested in testing session

**Review by Strohgelaender — APPROVED** [review]
tested on ts6, works and code looks good

**Review by florian-glombik — APPROVED** [review]
Tested on ts6 together with https://github.com/ls1intum/Orion/pull/94, worked fine

**Review by krusche — CHANGES_REQUESTED** [review]
I don't understand why the button "reload feedback" is necessary. 
In addition the second screenshot looks quite ugly. Why is the vertical button not at the right side any more?

**Comment by GunniBusch** [comment]
Fixed https://github.com/ls1intum/Orion/issues/93

**Pull request closed without merging**